### ISSUE TITLE
I know how we can write a basic variant of remarks in one, self-contained HTML-file

### ISSUE THREAD

**Issue by oleks** [DIKU-EDU/remarks#21]
That means, the logic will be written in JavaScript, but there isn't much logic to remarks, especially when little parsing logic is needed due to the availability of a DOM.

The benefit would be 0-entry for new TA's, and it will be trivial to send out HTML-versions to students or external examiners. The down-side is that there won't be a file structure to separate out concerns and avoid git merge conflicts if several TA's grade the same submission.

Let's talk if you are interested, else I will hack something up soon enough.

**Comment by kirkedal** [comment]
It sounds interesting, but what does basic hold? Is it just intended to be
used for checking missing tasks?

I agree that it can add some overhead to TAs that they have to install ghc
and other packages.

I will not have time to look at it before A0 of CompSys have to be
corrected, but could be interested later.

**Comment by oleks** [comment]
I have now used [html-remarks](https://github.com/oleks/html-remarks) for a single course and grown wiser. (Note, I did not have TAs.) html-remarks was a fairly nice, zero-install remarks UI for TA tasks (marking), but it was a poor teacher UI. For the students it could have been made prettier..

Today, I think that remarks as a DSL holds greater utility: **remarks as a DSL could have several front-ends, just as it today has several export formats**.

The DSL is explicitly designed to be git-friendly, while HTML and DOM manipulations were not. They can be hacked to be so, but this is riding a wave of hype, rather than having a neat, long-lasting solution. Riding this wave is OK to increase adoption, but it should probably not be the fundamental architecture of remarks.

Other than the UI benefits, [html-remarks](https://github.com/oleks/html-remarks) could have had the benefit that TAs (or students?) could contribute knowing only HTML, CSS, and TypeScript. Should remarks get an HTML+CSS+TypeScript front-end and export-format, they would probably still be able to do so! Furthermore, having a DSL for manipulating the remarks DSL, is probably better in the long-run too.
